_source/_posts/2022-07-21-jhipster-ci-cd.md

```diff
@@ -0,0 +1,865 @@
+---
+layout: blog_post
+title: "Continous integration and delivery for JHipster microservices"
+author: jimena-garbarino
+by: contractor
+communities: [devops,security,java]
+description: "How to set up continuous integration with CircleCI and continuous delivery with Spinnaker in a JHipster microservices architecture"
+tags: [java, ci, cd, jhipster]
+tweets:
+- ""
+- ""
+- ""
+image:
+type: awareness
+---
+<!--
+Intro
+What is CircleCI, CI
+What is Spinnaker, CD
+Workflow
+-->
+As requested by JHipster users, in this post we cover the basics of how to add Continous Integration and Delivery for a JHipster microservices architecture and Kubernetes as the target cloud deployment environment.
+
+In a few words, continuous integration is the practice of integrating code into a main branch of a shared repository early and often. Instead of integrating features at the end of a development cycle, code is integrated with the shared repository multiple times throughout the day. Each commit triggers automated tests, so issues are detected and fixed earlier and faster, improving team confidence and productivity. The chosen continuous integration platform is from CircleCI, a company founded in 2011 and headquartered in San Francisco. They offer a free cloud to test their services.
+
+Continuous delivery is the practice of releasing to production often in a fast, safe, and automatic way, allowing faster innovation and feedback loops. Its adoption requires the implementation of techniques and tools like Spinnaker, an open-source, multi-cloud, continuous delivery platform that provides application management and deployment features. The intersection between CI and CD is not always clear, but for this example, we assume CI produces and validates the artifacts and CD deploys them. The CI-CD workflow for the exploration of the proposed tools is illustrated below.
```

Comment:
```suggestion
Continuous delivery is the practice of releasing to production often in a fast, safe, and automated way, allowing faster innovation and feedback loops. Its adoption requires the implementation of techniques and tools like Spinnaker, an open-source, multi-cloud, continuous delivery platform that provides application management and deployment features. The intersection between CI and CD is not always clear, but for this example, we assume CI produces and validates the artifacts and CD deploys them. The CI-CD workflow for the exploration of the proposed tools is illustrated below.
```
"automated" has the nuance that the automation is intentional, not just mechanical...